#!/usr/bin/env python
PACKAGE = "kuka_iiwa_control_hbp"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("exp_ik_damp", double_t, 0, "Exponent coefficient for exp-damped IK",    6.0, -10.0,   1000.0)
gen.add("K_IK_lin", double_t, 0, "Closed-Loop Inverse Kinematic gain, position error",    0.0, -10.0,   1000.0)
gen.add("K_IK_ang", double_t, 0, "Closed-Loop Inverse Kinematic gain, orientation error",    0.0, -10.0,   1000.0)
gen.add("K_IK_2", double_t, 0, "Gain for secondary target of inverse kinematic",    0.0, -10.0,   1000.0)
gen.add("K_CTp", double_t, 0, "Computed-Torque Control proportional gain",    60.0, -10.0,   1000.0)
gen.add("K_CTd", double_t, 0, "Computed-Torque Control derivative gain",    20.0, -10.0,   1000.0)

exit(gen.generate(PACKAGE, "kuka_iiwa_control_hbp", "Controller"))